### ISSUE TITLE
Freeze after re-opening dialog with contact whose history was deleted

### ISSUE THREAD

**Issue by zak333** [siacs/Conversations#3054]
#### General information

* **Version:** 2.2.1
* **Device:** BQ Aquaris X5 Plus
* **Android Version:** LineageOS 14.1, Android 7.1.2
* **Server name:** jabber.ccc.de
* **Conversations source:** F-Droid

#### Steps to reproduce

1. Open chat with user X@jabber.de
2. Select Options: Remove history
3. Close this chat (either through the tick in step 2 or afterwards by swiping it away)
4. Select Add new chat
5. Click on user X@jabber.de

#### Expected result

The (empty) chat dialog should open again.

#### Actual result

Nothing happens. The contact shows the click-animation, then Conversations hangs for about a minute or so. Then I get the option to force close Conversations.
After a restart, the chat for contact X is there again, including all previous messages and I can try everything again with the same result.

#### Debug output

````
06-02 15:53:41.659 D/conversations(15885): ConversationsOverviewFragment.onStart()
06-02 15:53:41.662 D/conversations(15885): ConversationsOverviewFragment.onResume()
06-02 15:53:41.896 D/conversations(15885): initializing database...
06-02 15:53:41.896 D/conversations(15885): restoring accounts...
06-02 15:53:41.906 D/conversations(15885): restoring conversations...
06-02 15:53:41.909 D/conversations(15885): finished restoring conversations in 2ms
06-02 15:53:41.909 D/conversations(15885): restoring roster...
06-02 15:53:41.924 D/conversations(15885): stopped foreground service
06-02 15:53:41.924 D/conversations(15885): update unread count to 0
06-02 15:53:41.970 D/conversations(15885): restoring messages...
06-02 15:53:41.992 D/conversations(15885): finished restoring messages in 22ms
06-02 15:53:42.004 D/conversations(15885): EventReceiver ignored action android.net.conn.CONNECTIVITY_CHANGE
06-02 15:53:42.004 D/conversations(15885): app switched into foreground
06-02 15:53:42.058 D/conversations(15885): start merging phone contacts with roster
06-02 15:53:42.059 D/conversations(15885): finished merging phone contacts
06-02 15:53:42.125 D/conversations(15885): updated dynamic shortcuts
06-02 15:53:50.070 D/conversations(15885): ConversationsOverviewFragment.onPause()
06-02 15:53:51.795 D/conversations(15885): stopped foreground service
06-02 15:53:51.797 D/conversations(15885): me@jabber.ccc.de: socket was null during force close
06-02 15:53:51.797 D/conversations(15885): me@jabber.ccc.de: connecting
06-02 15:53:52.229 D/conversations(15885): Resolver: [Result{ip='146.255.57.229', hostame='jabberd.jabber.ccc.de', port=5222, directTls=false, authenticated=false, priority=10}, Result{ip='2a02:1b8:10:31::229', hostame='jabberd.jabber.ccc.de', port=5222, directTls=false, authenticated=false, priority=10}, Result{ip='146.255.57.229', hostame='jabberd.jabber.ccc.de', port=80, directTls=false, authenticated=false, priority=20}, Result{ip='2a02:1b8:10:31::229', hostame='jabberd.jabber.ccc.de', port=80, directTls=false, authenticated=false, priority=20}, Result{ip='2a02:1b8:10:31::229', hostame='jabberd-ipv6.jabber.ccc.de', port=5222, directTls=false, authenticated=false, priority=30}, Result{ip='null', hostame='jabberd-ipv6.jabber.ccc.de', port=5222, directTls=false, authenticated=false, priority=30}, Result{ip='146.255.57.229', hostame='jabberd-ipv4.jabber.ccc.de', port=5222, directTls=false, authenticated=false, priority=31}]
06-02 15:53:52.231 D/conversations(15885): me@jabber.ccc.de: using values from dns jabberd.jabber.ccc.de/146.255.57.229:5222 tls: false
06-02 15:53:52.536 D/conversations(15885): me@jabber.ccc.de: TLS connection established
06-02 15:53:52.568 D/conversations(15885): me@jabber.ccc.de/Conversations.stjt: Authenticating with SCRAM-SHA-1
06-02 15:53:52.774 D/conversations(15885): me@jabber.ccc.de: logged in
06-02 15:53:53.168 D/conversations(15885): me@jabber.ccc.de: not counting stanza(IqPacket). Not in smacks session.
06-02 15:53:53.171 D/conversations(15885): me@jabber.ccc.de: starting service discovery
06-02 15:53:53.187 D/conversations(15885): me@jabber.ccc.de: server caps came from cache
06-02 15:53:53.205 D/conversations(15885): me@jabber.ccc.de: stream management(3) enabled (resumable)
06-02 15:53:53.242 D/conversations(15885): me@jabber.ccc.de: Requesting block list
06-02 15:53:53.445 D/conversations(15885): me@jabber.ccc.de: successfully enabled carbons
06-02 15:53:53.448 D/conversations(15885): Received blocklist update from server
06-02 15:53:53.616 D/conversations(15885): AxolotlService (me@jabber.ccc.de): Bundle 4527576 in PEP was current
06-02 15:53:53.617 D/conversations(15885): AxolotlService (me@jabber.ccc.de): Announcing device 4527576
06-02 15:53:53.623 D/conversations(15885): me@jabber.ccc.de: online with resource Conversations.stjt
06-02 15:53:53.623 D/conversations(15885): me@jabber.ccc.de: fetching roster version 33e925ca582c53f94175567b8e8904e8df1bf957
06-02 15:53:53.627 D/conversations(15885): me@jabber.ccc.de: running mam query with=*, order=NORMAL, after=1527937527264474, catchup=true
06-02 15:53:53.628 D/conversations(15885): me@jabber.ccc.de sending csi//active
06-02 15:53:54.573 D/conversations(15885): me@jabber.ccc.de: retrieved own device list: [778346593, 4527576, 235543380]
06-02 15:53:54.591 D/conversations(15885): me@jabber.ccc.de: successfully purged offline messages
06-02 15:53:54.665 D/conversations(15885): me@jabber.ccc.de: finished mam after 1(0) messages. messages left=true count=398
06-02 15:53:54.665 D/conversations(15885): me@jabber.ccc.de: found 0 pending receipt requests
06-02 15:53:54.679 D/conversations(15885): AxolotlService (me@jabber.ccc.de): Received PEP device list [140757728] update from A@jabber.ccc.de, processing... 
06-02 15:53:54.853 D/conversations(15885): AxolotlService (me@jabber.ccc.de): Received PEP device list [2038141689, 1588376290] update from X@jabber.de, processing... 
06-02 15:53:54.882 D/conversations(15885): AxolotlService (me@jabber.ccc.de): Received PEP device list [1965554566] update from B@jabber.org, processing... 
06-02 15:53:54.887 D/conversations(15885): AxolotlService (me@jabber.ccc.de): Received PEP device list [778346593, 4527576, 235543380] update from me@jabber.ccc.de, processing... 
06-02 15:53:54.887 D/conversations(15885): me@jabber.ccc.de: ignoring duplicate own device id list
06-02 15:53:58.992 D/conversations(15885): ConversationsOverviewFragment.onStart()
06-02 15:53:58.993 D/conversations(15885): ConversationsOverviewFragment.onResume()
06-02 15:54:00.873 D/conversations(15885): ConversationsOverviewFragment.onPause()
06-02 15:54:02.355 D/conversations(15885): ConversationFragment.onAttach()
06-02 15:54:02.381 D/conversations(15885): reInit(hasExtras=true)
06-02 15:54:02.381 D/conversations(15885): trimming conversation
06-02 15:54:02.388 D/conversations(15885): scrolledToBottomAndNoPending=true
06-02 15:54:02.388 D/conversations(15885): jump to first unread message
06-02 15:54:02.414 D/conversations(15885): ConversationFragment.onBackendConnected()
06-02 15:54:02.486 D/conversations(15885): me@jabber.ccc.de: running mam query with=X@jabber.de, start=2018-06-02T11:05:28.303Z, end=2018-06-02T13:53:53.189Z, order=REVERSE, catchup=false
06-02 15:54:02.545 D/conversations(15885): me@jabber.ccc.de: finished mam after 0(0) messages. messages left=false count=0
06-02 15:54:02.545 D/conversations(15885): me@jabber.ccc.de: found 0 pending receipt requests

--- Here I removed the history and tried opening the chat dialog again.

06-02 15:54:13.999 D/conversations(15885): ConversationsOverviewFragment.onStart()
06-02 15:54:13.999 D/conversations(15885): ConversationsOverviewFragment.onResume()
06-02 15:54:15.448 D/conversations(15885): ConversationsOverviewFragment.onPause()

--- That's it, unfortunately nothing more.
````

I assume it might have something to do with the fact, that I restored a backup from 11/2017 recently which was a 1.x version. Probably something is corrupt in the database?

I would be happy to start over with a clean database, but would really like to keep my OMEMO keys if that's possible somehow?

**Comment by licaon-kter** [comment]
Encrypted chat with x@... or unencrypted? If unencrypted that's normal.

**Comment by zak333** [comment]
There were mostly encrypted messages in the history, yes. That's why I want to remove it.
But I can't imagine it's normal that Conversation freezes after re-opening a dialog with a contact.

**Comment by licaon-kter** [comment]
There were some issues with indexing that got fixed in 2.2.2 upadate and retest.

**Comment by zak333** [comment]
Confirmed. Version 2.2.2 fixed the problem. Thanks.
